lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,200 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const GcpStorage = require('@google-cloud/storage');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const ComputeClient = require('./ComputeClient');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class GcpClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(_.chain(this.settings).value());
+    this.storageClient = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+
+    return this.storageClient
+      .bucket(container)
+      .get();
+  }
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    const prefix = options ? options.prefix : null;
+    const queryOptions = {};
+    if (prefix) {
+      queryOptions.prefix = prefix;
+    }
+    queryOptions.autoPaginate = (options && options.autoPaginate) ? options.autoPaginate : true;
+
+    return Promise.try(() => {
+      return this.storageClient
+        .bucket(container)
+        .getFiles(queryOptions)
+        .then(results => {
+          const resultFiles = results[0];
+          const files = []; 
+          _.each(resultFiles, file => files.push(_       
+            .chain(file)       
+            .pick('name')       
+            .value()      
+          ));   
+          return files;
+        });
```

Comment:
Lines 57 thru 66 can be replaced with this 
```
.then(results => results instanceof Array? results.map((file) => _.get(file, 'name')): []);
```

Reply:
- We have to get the filenames from results[0] actually.
- Also, for other IaaS clients, like [AzureClient](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/rel-2017.T12b/lib/iaas/AzureClient.js#L61), we also pick lastModified, which I checkd is not getting used. Hence started sending files as an array of objects instead of array of names. I think let's keep this as an array of objects for future needs.

Reply:
It can just be like this. No need of returns when there is just one line.
```
.then(results => _.get(results, 0, []).map(file => ({name: file.name}));
```

Reply:
Done, please check and if everything is looks fine, please approve.